docs/homepage/blog/ospp_mid-term_report_210370741/index.md

```diff
@@ -0,0 +1,348 @@
+
+Following are the future work that are possible in this project.
+
+- Parallel loading and partial loading of datasets for supported datasets.
+- Add support for envs that are not supported by GymEnvs -> Flow and CARLA.
+- Add support for datasets in Flow and CARLA envs.
+- Add support for creating, storing and loading custom made datasets.
+- test-train split functionality for datasets.
+- cross validation and grid search.
+- build_with_dataset - make a particular algorithm based on the requirements of the env.
+- evaluator function that performs evaluation (can be on policy or off policy)
+- Metrics as hooks. Refer [Metrics](https://d3rlpy.readthedocs.io/en/v0.90/references/metrics.html)
+
+Refer the following [discussion](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/discussions/359) for more ideas that could be possible.
+
+### Time Planning
+
+| Date       | Goals |
+| :-----------: | :---------: |
+| 07/01 - 07/14 | Brainstorm various ideas that are possible for the implementation of RLDatasets.jl and finalize the key features. |
+| 07/15 - 07/20 | Made a basic julia wrapper for `d4rl` environments and add some tests |
+| 07/21 - 07/30 | Implement loading of `d4rl` and `d4rl-pybullet` datasets |
+| 07/31 - 08/06 | Implement loading of `Google Research DQN Replay Datasets` |
+| 08/07 - 08/14 | Implement loading of `RL Unplugged atari datasets`, setup the docs, add README.md. Make the package more user friendly. Make the **mid-term report** |
+| 08/15 - 08/30 | Add lazy loading and multi threaded loading support for `Google Research DQN Replay Datasets`. Add the rest of RL Unplugged datasets, polish the interface, finalize the structure of the codebase. Add examples and `register the package.` |
+| 09/01 - 09/15 | Add support for policy loading from [Benchmarks for Deep Off-Policy Evaluation](https://github.com/google-research/deep_ope) and implement an OPE method |
+| 09/16 - 09/30 | Test OPE in various datasets and publish benchmarks in RLDatasets.jl. Implement other features that makes the package more user friendly. Complete the **final-term report** |
+
+## 3. Implemented datasets
+
+### D4RL
+
+Added support for D4RL datasets with all features loaded in the returned type.
+
+Credits: https://github.com/rail-berkeley/d4rl
+
+```julia
+using ReinforcementLearningDatasets
+ds = dataset(
+        "hopper-medium-replay-v0";
+        repo="d4rl")
+samples = take!(ds, 2)
+```
+
+The type (`D4RLDataSet`) returned by `dataset` is an iterable that returns batches of data based on the requirement that is specified.
+
+### d4rl-pybullet
+
+Added support for datasets released in `d4rl-pybullet`. This enables testing the agents in complex environments without `Mujoco` license.
+
+Credits: https://github.com/takuseno/d4rl-pybullet
+
+```julia
+using ReinforcementLearningDatasets
+ds = dataset(
+        "hopper-bullet-mixed-v0";
+        repo="d4rl-pybullet",
+    )
+samples = take!(ds, 2)
```

Comment:
Could you print the result of a batch from julia-repl? That would help others understand the result more easily.

Reply:
it is very long. But I could print out the types for information